kymatio/scattering2d/frontend/torch_f.py

```diff
@@ -0,0 +1,178 @@
+            The maximum order of scattering coefficients to compute. Must be either
+            `1` or `2`. Defaults to `2`.
+        pre_pad : boolean, optional
+            controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to true, the software will assume the signal was
+            padded externally.
+
+        Attributes
+        ----------
+        J : int
+            logscale of the scattering
+        shape : tuple of int
+            spatial support (M, N) of the input
+        L : int, optional
+            number of angles used for the wavelet transform
+        max_order : int, optional
+            The maximum order of scattering coefficients to compute.
+            Must be either equal to `1` or `2`. Defaults to `2`.
+        pre_pad : boolean
+            controls the padding
+        Psi : dictionary
+            containing the wavelets filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        Phi : dictionary
+            containing the low-pass filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        M_padded, N_padded : int
+             spatial support of the padded input
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8
+
+        pre_pad is particularly useful when doing crops of a bigger
+         image because the padding is then extremely accurate. Defaults
+         to False.
+
+        """
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False):
+        super(Scattering2D, self).__init__(J, shape, max_order = max_order)
+        self.L, self.J, self.max_order = L, J, max_order
+        self.pre_pad = pre_pad
+        self.shape = shape
+        if 2 ** J > shape[0] or 2 ** J > shape[1]:
+            raise (RuntimeError('The smallest dimension should be larger than 2^J'))
+        self.build()
+
+    def build(self):
+        self.M, self.N = self.shape
+        self.modulus = Modulus()
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = Pad([(self.N_padded - self.N) // 2, (self.N_padded - self.N + 1) // 2, (self.M_padded - self.M) // 2,
+                       (self.M_padded - self.M + 1) // 2], [self.N, self.M], pre_pad=self.pre_pad)
+        self.subsample_fourier = SubsampleFourier()
+        self.filters = self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c])
```

Comment:
This loop does two things: convert from numpy to torch, and register the torch arrays. Should this be two different loops (or two different functions)? Maybe the former should happen in a `create_filters` function and the latter in a `register_filters` function?

Reply:
I think that's fine because one sees I had to trick to track the registered buffers. Those functions will be updated when torch will improve its buffer routines.

Reply:
Also another point is that this initialization will be different for 1/3-D Scattering. Ultimately, we would like this manner to instantiate the buffers to be identical for every scattering. This can't be done curretnly because we need more harmonization between 1-2-3 D.

Reply:
> I think that's fine because one sees I had to trick to track the registered buffers. 

I don't understand. What's the trick?

Reply:
storing each element of the array as a variable 'buffer_$i' where $i is incremented.